File: conformance/repo-lint/fixtures/violations/perl/missing_docstring.pl
#!/usr/bin/env perl
use strict;
use warnings;

# Test fixture: intentionally bad Perl code

# Subroutine without POD documentation (violates repo contract)
sub function_without_docstring {
    my ($param1, $param2) = @_;
    return $param1 + $param2;
}

# Perl::Critic: ProhibitExplicitReturnUndef (not auto-fixable)
sub bad_return {
    my $value = shift;
    return undef if !defined $value;  ## no critic (ProhibitExplicitReturnUndef) - Remove this to trigger violation
    return $value;
}

# Another function missing docs
sub another_function {
    print "test\n";
}

# Perl::Critic: RequireUseWarnings is satisfied above,
# but we can add other non-auto-fixable issues

1;

File: conformance/repo-lint/fixtures/violations/python/missing_docstring.py
"""Test fixture: intentionally bad Python code with missing docstrings."""

from __future__ import annotations


def function_without_docstring(param1, param2):
    return param1 + param2


class ClassWithoutDocstring:
    def method_without_docstring(self):
        pass


# Unused import (Ruff F401 - not auto-fixable as it might be intentional)
import os  # noqa: F401 - Remove this noqa to trigger violation
import sys


# Line too long (Ruff E501 - not auto-fixable as Black won't break it)
VERY_LONG_STRING_THAT_EXCEEDS_THE_LINE_LENGTH_LIMIT_OF_ONE_HUNDRED_TWENTY_CHARACTERS_AND_CANNOT_BE_AUTO_FIXED_BY_BLACK = (
    "value"
)


def _private_function_with_docs(x):
    """Example of a private function that IS documented (not a violation)."""
    return x * 2
